### PR TITLE
[5.2][Events] Use event classes for Actionlog plugins

### PR DESCRIPTION
**Pull request by Fedik**
### Summary of Changes

Update `actionlog` plugins to use `SubscriberInterface` and event classes.
And adding a couple missing events for User pw reset.

### Testing Instructions

Check that actionlog works as before.
Test User reset requests, should work as before.

### Actual result BEFORE applying this Pull Request
Works

### Expected result AFTER applying this Pull Request
Works

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed
- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Review by HLeithner — APPROVED** [review]

**Comment by exlemor** [comment]
I have tested this item :white_check_mark: successfully on d6ff121a15e1a6b1b4e99e29bc406b5096f3c186<br /><br />I was able to test this successfully.

I did an incorrect login which ended up in the action log, I did a full user reset which works perfectly (but no trace in the logs which I believe is standard).<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43639">issues.joomla.org/tracker/joomla-cms/43639</a>.</sub>

**Comment by rdeutz** [comment]
LGTM on Code Review

**Comment by Hackwar** [comment]
Thank you @Fedik for your contribution! 😊

File: plugins/actionlog/joomla/src/Extension/Joomla.php [hunk 1]
```diff
@@ -462,19 +514,21 @@ public function onExtensionAfterUpdate($installer, $eid)
...
-    public function onExtensionAfterSave($context, $table, $isNew): void
+    public function onExtensionAfterSave(Model\AfterSaveEvent $event): void
     {
+        $context = $event->getContext();
+        $table   = $event->getItem();
+        $isNew   = $event->getIsNew();
+
         $option = $this->getApplication()->getInput()->getCmd('option');
 
         if ($table->get('module') != null) {
```
Comment (Quy) [inline plugins/actionlog/joomla/src/Extension/Joomla.php:534]: Possibly due to this PR?

Edit a plugin.
Click Save button.
See PHP error log.

> PHP Warning:  Undefined property: Joomla\CMS\Table\Extension::$module in \plugins\actionlog\joomla\src\Extension\Joomla.php on line 536

Comment (Fedik) [reply plugins/actionlog/joomla/src/Extension/Joomla.php:534]: I do not think. The PR does not change the Table class.
Could be possible due to one of recent refactorings , for legacy get/set, or kind of that.
Better open new issue for this.

**Pull request merged**